components/Slider/src/MDCSlider.m

```diff
@@ -583,14 +587,25 @@ - (void)thumbTrackValueChanged:(__unused MDCThumbTrack *)thumbTrack {
   [self sendActionsForControlEvents:UIControlEventValueChanged];
   UIAccessibilityPostNotification(UIAccessibilityAnnouncementNotification, self.accessibilityValue);
   if (@available(iOS 10.0, *)) {
-    if (self.hapticsEnabled) {
+    if (self.hapticsEnabled && _previousValue != _thumbTrack.value){
+      BOOL valueCrossesAboveAnchor = (_previousValue < _thumbTrack.filledTrackAnchorValue &&
+                                      _thumbTrack.filledTrackAnchorValue <= _thumbTrack.value);
+      BOOL valueCrossesBelowAnchor = (_thumbTrack.value <= _thumbTrack.filledTrackAnchorValue &&
+                                      _thumbTrack.filledTrackAnchorValue < _previousValue));
+      BOOL crossesAnchor =
+          (!_firstValueChange) && (valueCrossesAboveAnchor ||
+                                   valueCrossesBelowAnchor);
       if (self.shouldEnableHapticsForAllDiscreteValues ||
           _thumbTrack.value == _thumbTrack.minimumValue ||
-          _thumbTrack.value == _thumbTrack.maximumValue) {
+          _thumbTrack.value == _thumbTrack.maximumValue || crossesAnchor) {
```

Comment:
Behaviorally-speaking we may want the slider to expose a mode for configuring whether it uses min/max haptics or anchor-based haptics. In the case linked to in the associated bug, for example, we do not want haptics to fire on min/max values because it does not have a semantic meaning in that context.

Reply:
@yarneo and I discussed this and were thinking that you would still generally want haptics and min/max (in addition to at the anchor) because this is standard for how apple has haptics on their sliders and because it also has some "meaning" in the sense that you know you have reached the end of the slider. In addition, we looked on other apps that have similar sliders for photo editing and they had haptics at the min and max as well. For these reasons and in order to keep it simpler we decided against having an option for just anchor haptics, but if we feel it would be better to split them I would be happy to.